app/javascript/components/explore/plotDataCache.js

```diff
@@ -0,0 +1,278 @@
+import { fetchCluster } from 'lib/scp-api'
```

Comment:
Since there's a _lot_ going on here, it would be great if each method had standard docstrings with params/returns/etc.

Reply:
I added a bunch more function-level comments.  I'm hesitant to go full docstring, because there are a *lot* of small methods that all pass around the same params, and to me, the readability suffers if  `@param cluster (string) - the cluster name` gets repeated everywhere. But I'll defer to you and Eric on what you think